src/public/Import-OriPsModule.ps1

```diff
@@ -0,0 +1,105 @@
+        6. Registers repository if not yet
+        7. Installs the module
+        8. Loads required version of the module
+
+    .PARAMETER Name
+        Specifies the exact names of modules to install from the online gallery. The module name must match the module name in the repository.
+        
+    .PARAMETER RequiredVersion
+        Specifies the exact version of a single module to install.
+    #>
+    [CmdLetBinding()]
+    param (
+        [parameter(Mandatory = $true, HelpMessage = "Exact name of the mmodule")]
+        [string] $Name,
+
+        [parameter(Mandatory=$true, HelpMessage="Minimum required module version")]
+        [Version] $RequiredVersion
+    )
+
+    # Required module is already imported
+    if (Test-GetModule -Name $Name -RequiredVersion $RequiredVersion) {
+        Write-Verbose "Module $Name is already imported.";
+        return;
+    }
+
+    # Required module is installed, but not imported
+    Import-Module -Name $Name -RequiredVersion $RequiredVersion -ea SilentlyContinue;
+
+    if (Test-GetModule -Name $Name -RequiredVersion $RequiredVersion) { 
+        Write-Verbose "Module $Name was imported from installed modules.";
+        return;
+    }
+
+    # Required module needs to be installed and impoted
+    "GlobalDev", "PackageManagementFeed" | Invoke-RegisterOriflameFeeds
+    Install-Module -Name $Name -RequiredVersion $RequiredVersion -Force;
+    Import-Module -Name $Name -RequiredVersion $RequiredVersion;
+}
+
+
+function Test-GetModule {
+    <#
+    .SYNOPSIS
+        Checks, whether specified module is installed.
+    #>
+    [CmdLetBinding()]
+    [OutputType([bool])]
+    param (
+        [parameter(Mandatory = $true, HelpMessage = "Exact name of the mmodule")]
+        [string] $Name,
+
+        [parameter(Mandatory=$true, HelpMessage="Minimum required module version")]
+        [Version] $RequiredVersion
+    )
+
+    $module = Get-Module -Name $Name -ea SilentlyContinue;
+
+    $null -ne $module -and $module.version -ge $RequiredVersion;
+}
+
+function Invoke-RegisterOriflameFeeds {
+    <#
+    .SYNOPSIS
+        Register required NuGet feeds from Oriflame artifact sources.
+    .NOTES
+        This cmdlet also install PowerShellGet module, if required.
+    #>
+    [CmdLetBinding()]
+    param (
+        [Parameter(ValueFromPipeline = $true, HelpMessage = "List of Oriaflame NuGet feeds to register (eg. GlobalDev, PackageManagementFeed).")]
+        [string[]] $feeds
+    )
+    begin {
+        if (!(Test-GetModule -Name PowerShellGet -RequiredVersion 2.2.3)) {
+            Write-Verbose "Updating PowerShellGet module to the latest version.";
+            Install-Module -Name PowerShellGet -Force;
+        }
+
+        $registeredFeeds = Get-PSRepository | Select-Object -Expand SourceLocation;
+    }
+    process {
+        foreach ($feed in $feeds) {
+            if ($registeredFeeds -like "*/$feed/*") {
+                continue;
+            }
+    
```

Comment:
Each function should be in a separate *.ps1 file. 
Each function Invoke-RegisterOriflameFeeds and Test-GetModule should have own unit test.
Is it an idea to keep Invoke-RegisterOriflameFeeds and Test-GetModule public or private?

Reply:
We need to keep everything in the single file, because usage of this script is following
```powershell
$url = 'https://raw.githubusercontent.com/Oriflame/PsRepositoryBootstrap/develop/src/public/Import-OriPsModule.ps1'

if (!(Get-Command -name Import-OriPsModule -ea SilentlyContinue)) {
    Invoke-WebRequest -useBasicParsing -uri $url | Invoke-Expression
}

Import-OriPsModule -Name MyModule -RequiredVersion 1.0.2
```
See the Readme for the details...

I think we can keep `Invoke-RegisterOriflameFeeds` and `Test-GetModule` public. They can be useful for someone maybe...